ui/selectors/selectors.js

```diff
@@ -381,6 +382,36 @@ export function getInternalAccountsSortedByKeyring(state) {
   });
 }
 
+/**
+ * Merges accounts with balances with each corresponding account data from internal accounts
+ *
+ * @param state - redux state
+ * @param accountsWithBalances - list of accounts with balances
+ * @returns merged accounts list with balances and internal account data
+ */
+export function mergeAccounts(state, accountsWithBalances) {
```

Comment:
I think we can have a more descriptive name here. Perhaps something like getAllAccounts?